### PR TITLE
feat: add advanced payment list foundation

### LINKED ISSUE #4518: [TASK] Display Refund Status in Payments Dashboard
## Summary
Enhance the payments dashboard to display refund status indicators (`REFUNDED` or `PARTIALLY REFUNDED`) when a payment has been refunded, instead of showing the original payment status (`SUCCEEDED`).

## Goal
When merchants view their payments list or payment details, they should immediately see if a payment has been refunded without navigating to the refunds dashboard. This improves UX and reduces confusion about payment states.

- Payments with full refunds should display `REFUNDED` status with orange color
- Payments with partial refunds should display `PARTIALLY REFUNDED` status with orange color
- Payments without refunds continue to show their original status (green for succeeded, red for failed)

### Data Model
The refund data is already available in the payment response:
```rescript
type order = {
  status: string,
  amount: float,
  refunds: array<refunds>,
  // ... other fields
}

type refunds = {
  refund_id: string,
  amount: float,
  status: string,
  // ... other fields
}
```

### Implementation Steps

**1. Create Refund Status Utility Function**

Location: `src/screens/Transaction/Order/HSwitchOrderUtils.res`

Add a new function to calculate refund status based on the order's refunds array:
```rescript
type refundStatus = 
  | NoRefund
  | PartiallyRefunded
  | FullyRefunded

let calculateRefundStatus = (order: order): refundStatus => {
  let totalRefundAmount = order.refunds
    ->Array.filter(refund => refund.status == "succeeded")
    ->Array.reduce(0.0, (acc, refund) => acc +. refund.amount)
  
  if totalRefundAmount == 0.0 {
    NoRefund
  } else if totalRefundAmount >= order.amount {
    FullyRefunded
  } else {
    PartiallyRefunded
  }
}
```

**2. Update Status Variant Type**

Location: `src/screens/Transaction/Order/HSwitchOrderUtils.res`

Add new status variants to the existing type:
```rescript
type statusVariant = 
  | Succeeded
  | PartiallyCaptured
  | Failed
  | Cancelled
  | Processing
  | RequiresCustomerAction
  | RequiresConfirmation
  | RequiresPaymentMethod
  | CancelledPostCapture
  | Refunded           // NEW
  | PartiallyRefunded  // NEW
```

**3. Update Payment List Status Display**

Location: `src/screens/Transaction/Order/OrderEntity.res`

Modify the `getCell` function for the `Status` column (around line 736):
```rescript
| Status => {
    let refundStatus = HSwitchOrderUtils.calculateRefundStatus(order)
    let (displayStatus, color) = switch (orderStatus, refundStatus) {
    | (_, FullyRefunded) => ("REFUNDED", LabelOrange)
    | (_, PartiallyRefunded) => ("PARTIALLY REFUNDED", LabelOrange)
    | (Succeeded, _) => ("SUCCEEDED", LabelGreen)
    | (Failed, _) => ("FAILED", LabelRed)
    | (Cancelled | CancelledPostCapture, _) => ("CANCELLED", LabelRed)
    | (Processing, _) => ("PROCESSING", LabelBlue)
    | (RequiresCustomerAction | RequiresConfirmation | RequiresPaymentMethod, _) => 
        (order.status->String.toUpperCase, LabelBlue)
    | _ => (order.status->String.toUpperCase, LabelLightGray)
    }
    Label({title: displayStatus, color: color})
  }
```

**4. Update Payment Detail Status Badge**

Location: `src/screens/Transaction/Order/OrderEntity.res`

Modify the `useGetStatus` hook or create a wrapper that considers refund status for the payment detail view.

### Color Coding
- **Refunded / Partially Refunded**: Orange (use existing `LabelOrange`)
- **Succeeded**: Green (existing `LabelGreen`)
- **Failed / Cancelled**: Red (existing `LabelRed`)
- **Processing / Requires Action**: Blue (existing `LabelBlue`)

### Data Flow
```
API Response (existing)
  └─> order.refunds = [{refund_id: "...", amount: 100.0, status: "succeeded"}]

Frontend Calculation (new)
  └─> calculateRefundStatus(order)
      └─> Sum successful refunds, compare to order amount
      └─> Returns: FullyRefunded | PartiallyRefunded | NoRefund

UI Rendering (modified)
  └─> Displays "REFUNDED" or "PARTIALLY REFUNDED" with orange color
```

### Edge Cases
1. **Multiple Partial Refunds**: Sum all successful refunds; show "REFUNDED" when total >= order amount
2. **Failed Refunds**: Only include refunds with status == "succeeded" in calculation
3. **Pending Refunds**: Pending/processing refunds should not affect status display yet
4. **Currency**: Assume amounts are in same currency (backend handles currency conversion)

### Testing Scenarios
1. Payment with no refunds → Shows original status
2. Payment with partial refund → Shows "PARTIALLY REFUNDED"
3. Payment with full refund → Shows "REFUNDED"
4. Multiple partial refunds totaling full amount → Shows "REFUNDED"
5. Failed/pending refunds → Shows original status

## Subtasks

- [ ] Add `refundStatus` type and `calculateRefundStatus` function in `HSwitchOrderUtils.res`
- [ ] Add `Refunded` and `PartiallyRefunded` variants to `statusVariant` type
- [ ] Update `getCell` for Status column in payment list to use refund status
- [ ] Update `useGetStatus` hook in payment detail view to use refund status
- [ ] Test with payments having no refunds, partial refunds, and full refunds
- [ ] Verify color coding matches specification (orange for refunded states)
- [ ] Test edge cases: multiple partial refunds, failed refunds, pending refunds
- [ ] Update any relevant unit tests

## Notes
- This is a frontend-only change; no backend API modifications needed
- The refund data is already returned in the payment API response
- Maintain backward compatibility for payments without refunds

### LINKED ISSUE #5080: Add advanced payment list foundation
## Summary

Prepare the hidden foundation for the advanced payment list flow without enabling visible UI behavior.

## Scope

- Add payment list analytics/OpenSearch field mappings and table metadata.
- Add source-control and export support primitives for the advanced payment list flow.
- Add dormant transaction-view helpers for advanced payment activity cards.

## Rollout

This is the foundation PR. Visible wiring and user-facing behavior will be enabled in the follow-up PR.

### PR DESCRIPTION
**Pull request by VenuMadhav2541**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Adds the hidden foundation for the advanced payment list flow without enabling user-facing rendering in this PR.

Changes include:
- Payment analytics/OpenSearch field mappings, CSV metadata, and advanced payment list table metadata.
- Typed payment list source primitives and reusable source-control UI support.
- Dormant transaction-view helpers for advanced payment activity cards.
- Optional table customization support for marking newly added columns while preserving existing behavior by default.

This PR does not wire the feature into the Payments page, does not enable a route or feature flag, and does not change visible behavior by itself. The follow-up PR will wire the UI and make the feature available.

## Motivation and Context

The advanced payment list work is being split into a foundation PR and a visible feature PR to reduce review load and keep rollout risk controlled. This foundation PR lets reviewers validate the data mapping, types, reusable helpers, and non-rendered components independently before the follow-up PR activates the full experience.

## How did you test it?

- Ran `npm run re:format`
- Ran `git diff --check`
- Ran `npm run re:build`

## Where to test it?

- [x] INTEG
- [ ] SANDBOX
- [ ] PROD

## Backend Dependency

- [ ] Yes
- [x] No

Backend PR URL:

## Feature Flag

- [ ] New feature flag added
- [ ] Existing feature flag updated
- [x] No feature flag changes

Feature flag name(s):

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/components/NewFeatureTag.res [hunk 1]
```diff
@@ -0,0 +1,10 @@
+@react.component
+let make = (~className="", ~description="") => {
+  let tag =
+    <span className={`inline-flex shrink-0 items-center align-middle ${className}`}>
+      <TagBinding text="NEW" color=Primary variant=Subtle shape=Squarical size=Xs />
+    </span>
+  description->LogicUtils.isNonEmptyString
+    ? <ToolTip description toolTipFor=tag toolTipPosition=ToolTip.Top />
+    : tag
```
Comment (susrithasabbini) [inline src/components/NewFeatureTag.res]: RenderIf

File: src/screens/Transaction/Order/PaymentListSourceControls.res [hunk 2]
```diff
@@ -0,0 +1,92 @@
+module SourceTabs = {
+  let getPaymentListSourceLabel = (source: OrderTypes.paymentListSource) => (source :> string)
```
Comment (susrithasabbini) [inline src/screens/Transaction/Order/PaymentListSourceControls.res]: open OrderTypes at the top

File: src/screens/Transaction/Order/PaymentListSourceControls.res [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  ]
+
+  let getPaymentListSourceFromLabel = value =>
+    paymentListSources->Array.find(source => source->getPaymentListSourceLabel == value)
+
+  let getPaymentListSourceDescription = source =>
+    switch source {
+    | OrderTypes.Normal => "Standard payments list."
+    | OrderTypes.Advanced => "Advanced payment list with expanded search, filters, columns, and CSV export."
+    }
```
Comment (susrithasabbini) [inline src/screens/Transaction/Order/PaymentListSourceControls.res]: Move to utils

File: src/screens/TransactionViews/TransactionViewUtils.res [hunk 4]
```diff
@@ -9,12 +10,41 @@ let paymentViewsArray: array<viewTypes> = [
...
 
 let disputeViewsArray: array<viewTypes> = [All, Succeeded, Failed, Pending]
 
 let payoutViewsArray: array<viewTypes> = [All, Succeeded, Failed, Cancelled, Expired, Reversed]
 
+let getTransactionViewGridClass = viewsCount =>
+  viewsCount >= 6
+    ? "grid lg:grid-cols-6 md:grid-cols-3 sm:grid-cols-2 grid-cols-1 gap-4"
+    : "grid lg:grid-cols-4 md:grid-cols-3 sm:grid-cols-2 grid-cols-1 gap-4"
+
```
Comment (susrithasabbini) [inline src/screens/TransactionViews/TransactionViewUtils.res]: This can be in the main component instead of utils, since it is the class

**Review by susrithasabbini — DISMISSED** [review]

**Comment by XyneSpaces** [comment]
[should-fix] Verify PCI compliance for `card_last_4` field in analytics

The new `card_last_4` field in `analytics_row` type stores sensitive card data. Please confirm:
1. This value is properly masked/tokenized by the backend before reaching analytics
2. Access is logged and auditable
3. The field is not exposed in any public API responses without proper authorization

If this is already handled by the backend data pipeline, please disregard. Otherwise, consider whether this field should be behind additional access controls or marked with PCI-handling annotations.

**Comment by XyneSpaces** [comment]
:warning: `card_last_4` and `card_network` fields in analytics lack PII handling

The analytics type definitions add `card_last_4` and `card_issuer` to the analytics row type. These are sensitive payment card fields that may have compliance implications (PCI-DSS) if logged or stored without proper handling.

Ensure these fields are:
1. Masked/encrypted in transit and at rest
2. Not logged in plaintext in application logs
3. Subject to appropriate data retention policies

**Comment by XyneSpaces** [comment]
## Automated Code Review Results

**Tier:** Foundation/Future-Proofing PR

### Findings Summary

**✅ Well-Structured Foundation PR**
This PR establishes the type foundations and UI components for the advanced payment list feature without wiring it to production paths. Good separation of concerns.

### Review Notes

**🎯 Architecture Assessment**

The modular approach is sound:
- `NewPaymentListTypes.res` — Core type definitions
- `NewPaymentListUtils.res` — Pure utility functions  
- `PaymentListSourceControl.res` — Reusable UI component
- `TransactionViewHelpers.res` — Dormant helpers for future wiring

This allows iterative development without impacting the current payments page.

**🔍 Noticing Existing Review Feedback**
PR already has helpful comments from @susrithasabbini:
1. Use `RenderIf` component for conditional rendering
2. Open `OrderTypes` module at top-level for cleaner access
3. Move helper functions to appropriate `*Utils.res` files
4. Keep styling classes in components rather than utils

**💡 Additional Suggestions:**

**In `PaymentListSourceControls.res` (line ~7):**
```rescript
let getPaymentListSourceLabel = (source: OrderTypes.paymentListSource) => (source :> string)
```

As noted in the existing review, consider opening the module:
```rescript
open OrderTypes
let getPaymentListSourceLabel = (source: paymentListSource) => (source :> string)
```

**For CSV Export Mappings (`AnalyticsUtils.res`):**
The new fields (card_last_4, card_network, etc.) are well-chosen for analytics. Ensure the OpenSearch indices are updated before this feature goes live.

**⚠️ Pre-Merge Checklist:**
- [ ] Address susrithasabbini's review comments
- [ ] Verify all new types have corresponding decoder/encoder functions
- [ ] Ensure TransactionViewHelpers.res functions have unit tests before they're wired to UI

**Overall Assessment:** Solid foundation work. Address the existing review feedback before merging. The modular structure will enable smooth iterative development.

**Comment by XyneSpaces** [comment]
## Review Summary

**PR #5081: feat: add advanced payment list foundation**

This PR establishes the foundation for an advanced payment list feature with OpenSearch integration. Review focuses on the structural additions to PaymentInterface types.

### Findings

**:mag: Naming consistency for new analytics fields**

The new fields added to `analytics_row` in `PaymentInterfaceAnalyticsTypes.res` follow snake_case convention appropriately. However, `routing_approach` may be ambiguous - verify this aligns with existing terminology in analytics exports.

```rescript
// Lines 56-64 additions:
card_last_4: string,
card_network: string,
card_issuer: string,
refunds_status: string,
refunds_count: int,
dispute_status: string,
dispute_count: int,
routing_approach: string,  // Verify naming consistency
```

**:mag: Consider field optionality**

Several new fields (card_last_4, card_network, card_issuer) may not always have values. Consider if these should be `option<string>` rather than required strings to avoid empty string sentinel values.

```rescript
// Current - all required strings
card_last_4: string,

// Suggested - optional for fields that may not exist
card_last_4?: string,
```

**Overall:** This appears to be a foundation PR (as noted in description). The type additions are reasonable for analytics expansion. Ensure follow-up PRs include:
- Actual UI wiring for the advanced list view
- OpenSearch query implementation
- CSV export functionality using the metadata types

**Comment by XyneSpaces** [comment]
## Review Summary

**Classification:** Control Center — Type definitions for OpenSearch payment analytics
**Risk:** Low

:white_check_mark: No blocking issues · Already reviewed by @susrithasabbini

---

### Findings

This PR adds foundational type definitions for advanced payment list functionality. The changes are purely additive to type interfaces:

1. **`PaymentInterfaceAnalyticsTypes.res`** — 8 new fields for OpenSearch analytics row types (card details, refund/dispute status, routing approach)

2. **`PaymentInterfaceTypes.res`** — 11 new optional fields on the `order` type

---

### Observations

- Field naming follows existing conventions (snake_case for analytics, camelCase for order types)
- Optional fields (`?`) are appropriate for backward compatibility with existing API responses
- The split between `analytics_row` (always-present fields) and `order` (optional fields) maintains consistency with the existing type hierarchy

Since @susrithasabbini has already provided detailed inline feedback on structural concerns, no additional review comments are warranted on these type definitions.

**Comment by XyneSpaces** [comment]
:bulb: Missing analytics event for source tab switching

The `PaymentListSourceControls.res` file handles tab switching between Normal and Advanced payment list sources, but there's no Mixpanel/analytics event fired when users switch sources. This makes it impossible to track adoption of the advanced payment list feature.

**Suggestion**: Add an analytics event when `setSource` is called, capturing:
- The source being switched to (Normal/Advanced)
- Whether the advanced feature is enabled for the merchant

This will help product teams understand feature adoption and guide future improvements.

**Comment by XyneSpaces** [comment]
## Code Review Findings

### Review of PR #5081: feat: add advanced payment list foundation

This PR adds infrastructure for an advanced payment list feature (foundation work, not yet wired to UI). Here are my findings:

#### [should-fix] Analytics fields use non-optional string types
**File:** `src/Interface/PaymentInterface/PaymentInterfaceTypes/PaymentInterfaceAnalyticsTypes.res`
**Line:** 59-66

All 8 new fields are typed as mandatory `string` or `int`:
```rescript
card_last_4: string,
card_network: string,
...
```

**Concern:** If the backend OpenSearch index doesn't have these fields populated for historical records or certain payment methods (e.g., wallets don't have `card_last_4`), deserialization will fail.

**Suggestion:** Use option types for fields that may not always be present:
```rescript
card_last_4: option<string>,
card_network: option<string>,
```

---

#### [should-fix] Feature flag gating not present
**Observation:** The PR adds significant new functionality but shows no feature flag gates in the visible diff.

**Concern:** Since this is "foundation" code that may not be complete, merging without feature flags risks exposing partial functionality or breaking existing flows if the types are consumed before the UI is ready.

**Suggestion:** Add feature flag checks (e.g., `useFeatureFlag` or similar) around:
1. New OpenSearch query fields
2. New table column renderers  
3. Type consumption sites

---

#### [nit] Refunds and disputes as flat fields
**File:** `src/Interface/PaymentInterface/PaymentInterfaceTypes/PaymentInterfaceAnalyticsTypes.res`
**Line:** 62-65

```rescript
refunds_status: string,
refunds_count: int,
dispute_status: string,
dispute_count: int,
```

**Suggestion:** Consider nesting related fields for clarity:
```rescript
refunds: {
  status: string,
  count: int
},
```

---

#### [should-fix] Missing docstrings for new types
**File:** Multiple files with new types

New type definitions lack docstrings explaining their purpose and intended usage.

**Suggestion:** Add ReScript docstrings (`/** */`) for:
- `PaymentListSourcePrimitives.res` exported types
- New OpenSearch mapping types
- Helper functions in `TransactionViewHelpers.res`

---

### Pre-existing Issues (not introduced by this PR)
None observed in the diff scope.

### Summary
The foundation structure looks reasonable, but the mandatory types and lack of feature flagging introduce risk. Recommend addressing the option types before merge, especially given this interfaces with external data (OpenSearch).

**Comment by XyneSpaces** [comment]
## Code Review Findings

**Skill:** control-center-review

### Findings

:warning: **PCI-sensitive card data fields in analytics type**

The `analytics_row` type adds `card_last_4`, `card_network`, and `card_issuer` fields. While `card_last_4` is generally considered non-sensitive (first 6 and last 4 are safe to log/display per PCI-DSS), verify:

1. These fields are populated from already-tokenized/redacted data in the backend, not raw card payloads
2. The analytics export/reporting functionality doesn't inadvertently expose these in ways that could be combined with other data points

```rescript
card_last_4: string,
card_network: string,
card_issuer: string,
```

Consider whether `card_last_4` should be `option<string>` to handle non-card payments gracefully without empty string defaults.

---

:bulb: **Inconsistent optionality pattern for new fields**

The PR adds fields that appear nullable in spirit (refunds_status, dispute_status, routing_approach) but are typed as plain `string`. Compare to existing optional fields in the codebase — if these can be absent in the API response, type them as `option<string>`.

---

:mag: **Naming consistency: snake_case vs camelCase**

The new fields follow `snake_case` (`card_last_4`, `refunds_status`) which matches the existing `attempt_count`, `connector_label` pattern in this type. Confirmed consistent with existing conventions.

---

**Summary:** Foundation work for advanced payment list looks solid. Verify PCI compliance for card data exposure in analytics contexts, and consider optionality for fields that may be absent in some payment flows.

**Comment by XyneSpaces** [comment]
[blocking] New fields added to `analytics_row` type without optional markers

The 8 new fields added to `analytics_row` (`card_last_4`, `card_network`, `card_issuer`, `refunds_status`, `refunds_count`, `dispute_status`, `dispute_count`, `routing_approach`) are declared as required `string`/`int` types. If the backend API returns null or omits these fields, deserialization will fail.

**Fix**: Make these fields optional:
```rescript
card_last_4?: string,
card_network?: string,
...
```

Verify that all fields added in `PaymentInterfaceAnalyticsTypes.res:59-66` are marked optional.

**Comment by XyneSpaces** [comment]
_Note: This is a test call to check if issue comments are accessible. Not an actual comment._

**Comment by XyneSpaces** [comment]
:warning: **New components lack feature flag gating**

The new advanced payment list components are not protected by any feature flag:
- `PaymentsTableAdvancedListUtils.res` (159 lines)
- `PaymentsTableAdvancedListTypes.res` (20 lines)
- `CsvMetadata.res` (212 lines)
- `SourceControl.res` (111 lines)
- `TransactionViewHelpers.res` (88 lines)

Without feature flag protection, these features will be immediately available to all merchants without gradual rollout capability. Consider adding a feature flag like `advancedPaymentList` or `paymentListOpensearch` in `FeatureFlagUtils.res` and gating the new module imports/renders behind it.

**Comment by XyneSpaces** [comment]
:warning: **Foundation code not behind feature flag risks premature exposure**

The PR adds 30 files of "foundation" code for an advanced payment list feature that's "not yet wired into the Payments page." However, I don't see any feature flag gating the new exports (`AdvancedPaymentListMetadata`, `AdvancedPaymentListTableColumns`, `PaymentListSourceControls`, etc.).

Without feature flags:
1. Partial/incomplete code could be accidentally imported and used
2. The bundle size increases for all users regardless of feature readiness
3. Future PRs that "wire" the feature can't easily control rollout

Please add feature flag checks (e.g., `useFeatureFlag` hook or compile-time flags) around:
- The new `paymentListSource` type exports
- The new filter types and constants in `OrderUIUtils`
- The new UI control components

This allows gradual rollout and quick disable if issues arise.

```suggestion
// In files exposing new functionality:
let useAdvancedPaymentListEnabled = () => {
  useFeatureFlag(~key="advanced_payment_list")
}
```

**Review by susrithasabbini — DISMISSED** [review]

File: src/components/NewFeatureTag.res [hunk 5]
```diff
@@ -0,0 +1,14 @@
+@react.component
+let make = (~className="", ~description="") => {
+  let hasDescription = description->LogicUtils.isNonEmptyString
+  let tag =
+    <span className={`inline-flex shrink-0 items-center align-middle ${className}`}>
+      <TagBinding text="NEW" color=Primary variant=Subtle shape=Squarical size=Xs />
+    </span>
+  <>
+    <RenderIf condition=hasDescription>
+      <ToolTip description toolTipFor=tag toolTipPosition=ToolTip.Top />
```
Comment (kanikabansal08) [inline src/components/NewFeatureTag.res]:
```suggestion
      <ToolTip description toolTipFor=tag toolTipPosition=Top />
```

File: src/screens/Transaction/Order/OrderTypes.res [hunk 6]
```diff
@@ -170,6 +180,130 @@ type optionObj = {
...
+  card_last_4: array<string>,
+  active_attempt_id: array<string>,
+  merchant_connector_id: array<string>,
+  refunds_status: array<string>,
+  dispute_status: array<string>,
+  routing_approach: array<string>,
+  card_issuer: array<string>,
+}
+
+type filter = [
```
Comment (kanikabansal08) [inline src/screens/Transaction/Order/OrderTypes.res:206]: we have these types in ordersUIUtils right? why are we maintaining a similar list here?

File: src/screens/Transaction/Order/OrderEntity.res [hunk 7]
```diff
@@ -462,6 +653,64 @@ let useGetStatus = order => {
...
+          toolTipFor={<div className="w-40 overflow-hidden">
+            <div className="flex items-center gap-1 whitespace-nowrap">
+              {activityTags
+              ->Array.map(tag =>
+                <TagBinding text=tag.label color=Primary variant=Subtle shape=Squarical size=Xs />
+              )
+              ->React.array}
+            </div>
+          </div>}
+          toolTipPosition=ToolTip.Top
```
Comment (kanikabansal08) [inline src/screens/Transaction/Order/OrderEntity.res]:
```suggestion
          </div>}
          toolTipPosition=Top
```

File: src/screens/Transaction/Order/OrderEntity.res [hunk 8]
```diff
@@ -462,6 +653,64 @@ let useGetStatus = order => {
   }
 }
 
+let formatActivityCount = (count, label) => {
```
Comment (kanikabansal08) [inline src/screens/Transaction/Order/OrderEntity.res:656]: can we please make a generic function to pluralise. this has been repeated in multiple places of codebase.

File: src/screens/Transaction/Order/OrderEntity.res [hunk 9]
```diff
@@ -314,6 +314,51 @@ let defaultColumns: array<colType> = [
...
   Modified,
 ]
+
+let openSearchDefaultColumns: array<colType> = [
+  PaymentId,
+  Connector,
+  ProfileId,
+  Amount,
+  Status,
+  Activities,
```
Comment (kanikabansal08) [inline src/screens/Transaction/Order/OrderEntity.res]: why Activities is in both default and new cols? line number 339

File: src/screens/Transaction/Order/OrderEntity.res [hunk 10]
```diff
@@ -370,6 +415,141 @@ let allColumnsV2 = [
...
+let csvHeaders =
+  openSearchCsvColumns->Array.map(column => (
+    column->getOpenSearchCsvKey,
+    column->getOpenSearchCsvHeader,
+  ))
+
+let mapOrderDictToCsvRow = (dict: Dict.t<JSON.t>) =>
+  openSearchCsvColumns
+  ->Array.map(column => (column->getOpenSearchCsvKey, getOpenSearchCsvValue(dict, column)))
+  ->Dict.fromArray
```
Comment (kanikabansal08) [inline src/screens/Transaction/Order/OrderEntity.res]: use `getJsonFromArrayOfJson`

File: src/screens/Transaction/Order/OrderEntity.res [hunk 11]
```diff
@@ -462,6 +653,64 @@ let useGetStatus = order => {
   }
 }
 
+let formatActivityCount = (count, label) => {
+  let suffix = count == 1 ? "" : "S"
+  `${count->Int.toString} ${label}${suffix}`
+}
+
+type activityTag = {label: string}
```
Comment (kanikabansal08) [inline src/screens/Transaction/Order/OrderEntity.res]: move the type at top

File: src/screens/Transaction/Order/PaymentListSourceControls.res [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+            value
+            variant=Boxed
+            size=Md
+            disabled=isDisabled
+            className="min-w-20 justify-center">
+            <ToolTip
+              description={item->getPaymentListSourceDescription}
+              toolTipFor={<span> {(item :> string)->React.string} </span>}
+              toolTipPosition=ToolTip.Top
+            />
```
Comment (kanikabansal08) [inline src/screens/Transaction/Order/PaymentListSourceControls.res]:
```suggestion
              toolTipPosition=Top
            />
```

File: src/screens/Transaction/Order/PaymentListSourceControls.res [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+          </TabsBinding.Trigger>
+        })
+        ->React.array}
+      </TabsBinding.List>
+    </TabsBinding>
+  }
+}
+
+module ExportButton = {
```
Comment (kanikabansal08) [inline src/screens/Transaction/Order/PaymentListSourceControls.res]: why are we making a export button like this? we can use existing button component and pass props.

File: src/screens/Transaction/Order/OrderEntity.res [hunk 14]
```diff
@@ -462,6 +653,64 @@ let useGetStatus = order => {
...
+
+let getActivityTags = (order: order) => {
+  let refundsCount = order.refunds_count->Option.mapOr(order.refunds->Array.length, count => count)
+  let disputeStatus = order.dispute_status->Option.getOr("")
+  let disputesCount =
+    order.dispute_count->Option.mapOr(
+      order.disputes->isNonEmptyArray
+        ? order.disputes->Array.length
```
Comment (kanikabansal08) [inline src/screens/Transaction/Order/OrderEntity.res]: we can use if else, 2 ternaries doesn't make this clear.

File: src/utils/LogicUtils.res [hunk 15]
```diff
@@ -587,6 +587,15 @@ let getTitle = name => {
...
 
+let pluralize = (~count, ~singular, ~plural=?) => {
+  let pluralValue = plural->Option.getOr(`${singular}s`)
+  if count == 1 {
+    singular
+  } else {
+    pluralValue
+  }
+}
+
```
Comment (kanikabansal08) [inline src/utils/LogicUtils.res]:
```suggestion
let pluralize = (~count, ~singular, ~plural=?) =>
  count == 1
    ? singular
    : plural->Option.getOr(`${singular}s`)
```

File: src/screens/Transaction/Order/OrderEntity.res [hunk 16]
```diff
@@ -462,6 +653,60 @@ let useGetStatus = order => {
   }
 }
 
+let formatActivityCount = (count, label) => {
+  `${count->Int.toString} ${pluralize(~count, ~singular=label, ~plural=`${label}S`)}`
+}
+
+let getActivityTags = (order: order) => {
+  let refundsCount = order.refunds_count->Option.mapOr(order.refunds->Array.length, count => count)
```
Comment (kanikabansal08) [inline src/screens/Transaction/Order/OrderEntity.res]: check if getor can we used, identity map is redundant
```suggestion
  let refundsCount = order.refunds_count->Option.getOr(order.refunds->Array.length)
```

File: src/screens/Transaction/Order/OrderEntity.res [hunk 17]
```diff
@@ -462,6 +653,60 @@ let useGetStatus = order => {
...
+  let disputeStatus = order.dispute_status->Option.getOr("")
+  let fallbackDisputesCount = if order.disputes->isNonEmptyArray {
+    order.disputes->Array.length
+  } else if disputeStatus->isNonEmptyString {
+    1
+  } else {
+    0
+  }
+  let disputesCount = order.dispute_count->Option.mapOr(fallbackDisputesCount, count => count)
```
Comment (kanikabansal08) [inline src/screens/Transaction/Order/OrderEntity.res]:
```suggestion
 let disputesCount = order.dispute_count->Option.getOr(fallbackDisputesCount) 
```

File: src/screens/Transaction/Order/OrderEntity.res [hunk 18]
```diff
@@ -462,6 +653,60 @@ let useGetStatus = order => {
   }
 }
 
+let formatActivityCount = (count, label) => {
+  `${count->Int.toString} ${pluralize(~count, ~singular=label, ~plural=`${label}S`)}`
```
Comment (kanikabansal08) [inline src/screens/Transaction/Order/OrderEntity.res]: do we need to pass ~plural=`${label}S won't this be covered in default case?

**Review by kanikabansal08 — APPROVED** [review]

**Review by susrithasabbini — APPROVED** [review]
